dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/communityManager/SingleDataRequestsTest.kt

```diff
@@ -0,0 +1,259 @@
+            requestControllerApi.patchDataRequest(storedDataRequestId, RequestStatus.resolved)
+        }
+        assertEquals("Client error : 403 ", clientException.message)
+    }
+
+    @Test
+    fun `query data requests with various filters and assert that the expected results are being retrieved`() {
+        val requestA = SingleDataRequest(
+            companyIdentifier = generateRandomIsin(),
+            frameworkName = SingleDataRequest.FrameworkName.lksg,
+            listOfReportingPeriods = listOf("2022"),
+        )
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+        requestControllerApi.postSingleDataRequest(requestA)
+
+        val specificPermId = System.currentTimeMillis().toString()
+        val requestB = SingleDataRequest(
+            companyIdentifier = specificPermId,
+            frameworkName = SingleDataRequest.FrameworkName.sfdr,
+            listOfReportingPeriods = listOf("2021"),
+        )
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Admin)
+        val req2 = requestControllerApi.postSingleDataRequest(requestB).first()
+        requestControllerApi.patchDataRequest(UUID.fromString(req2.dataRequestId), RequestStatus.resolved)
+
+        val allDataRequests = requestControllerApi.getDataRequests()
+        val lksgDataRequests = requestControllerApi.getDataRequests(
+            dataType = RequestControllerApi.DataTypeGetDataRequests.lksg,
+        )
+        val reportingPeriod2021DataRequests = requestControllerApi.getDataRequests(reportingPeriod = "2021")
+        val resolvedDataRequests = requestControllerApi.getDataRequests(requestStatus = RequestStatus.resolved)
+        val specificPermIdDataRequests = requestControllerApi.getDataRequests(
+            dataRequestCompanyIdentifierValue = specificPermId,
+        )
+        val specificUsersDataRequests = requestControllerApi.getDataRequests(userId = UPLOADER_USER_ID)
+
+        val allQueryResults = listOf(
+            allDataRequests, lksgDataRequests, reportingPeriod2021DataRequests,
+            resolvedDataRequests, specificPermIdDataRequests, specificUsersDataRequests,
+        )
+
+        allQueryResults.forEach { storedDataRequestsQueryResult ->
+            assertTrue(storedDataRequestsQueryResult.isNotEmpty())
+        }
+
+        assertTrue(allDataRequests.size > 1)
+        assertTrue(lksgDataRequests.all { it.dataType == StoredDataRequest.DataType.lksg })
+        assertTrue(reportingPeriod2021DataRequests.all { it.reportingPeriod == "2021" })
+        assertTrue(resolvedDataRequests.all { it.requestStatus == RequestStatus.resolved })
+        assertTrue(specificPermIdDataRequests.all { it.dataRequestCompanyIdentifierValue == specificPermId })
+        assertTrue(specificUsersDataRequests.all { it.userId == UPLOADER_USER_ID })
+    }
+
+    @Test
+    fun `query the data requests as an uploader and assert that it is forbidden`() {
```

Comment:
Why only for uploader?

Reply:
Because it is the next "lower" level than admin.
I thought that if an upload is already forbidden from doing this, then a reader is also forbidden.
The other way around not always!
What do you think?

Reply:
Should be fine. Lets not make to complicated and leave it as it is. Closing this